weblate/glossary/tasks.py

```diff
@@ -40,6 +48,29 @@ def sync_glossary_languages(pk: int, component: Component | None = None) -> None
         component.create_translations_task()
 
 
+@app.task(trail=False, autoretry_for=(Project.DoesNotExist, WeblateLockTimeoutError))
+def cleanup_stale_glossaries(project: int | Project) -> None:
+    if isinstance(project, int):
+        project = Project.objects.get(pk=project)
+
+    languages_in_non_glossary_components: set[int] = set(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=False
+        ).values_list("language_id", flat=True)
+    )
+
+    glossary_translations = prefetch_stats(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=True
+        ).exclude(language__id__in=languages_in_non_glossary_components)
+    )
+    for glossary in glossary_translations:
+        if glossary.stats.translated == 0:
+            glossary.remove(get_anonymous())
```

Comment:
Because of this, some tests are failing. But I think it merely unveils the issue somewhere else because it is trying to remove stale glossaries just from the `create_glossary` call what sounds wrong.

Reply:
that makes sense, we can execute the clean when deleting translations only